R/utilities-units.R

```diff
@@ -21,11 +21,16 @@ toBaseUnit <- function(quantityOrDimension, values, unit, molWeight = NULL) {
   validateIsNumeric(molWeight, nullAllowed = TRUE)
   unit <- enc2utf8(unit)
   dimension <- quantityOrDimension
+  dimensionTask <- getNetTask("DimensionTask")
+
   if (isOfType(quantityOrDimension, Quantity)) {
     dimension <- quantityOrDimension$dimension
   }
-  values <- c(values)
-  dimensionTask <- getNetTask("DimensionTask")
+
+  if(all(is.na(molWeight))){

```

Comment:
is.na(NULL) returns an empty array and if(is.na(NULL)) crashes
all(is.na(NULL) returns TRUE, so does all(is.na(NA)) 
We need to assign null after that and we are good to go